libvast/src/partition_synopsis.cpp

```diff
@@ -142,7 +183,8 @@ caf::error unpack_(
     synopsis_ptr ptr;
     if (auto error = unpack(*synopsis, ptr))
       return error;
-    if (qf.is_standalone_type())
+    // We mark type-level synopses by using an empty string as name.
+    if (qf.name().empty())
```

Comment:
This was changed with the type system refactoring; `is_standalone_type` supersedes that hack.

Reply:
I'm not sure that makes sense: why should the encoding of type vs. field synopses in the partition_synopsis flatbuffer be a concern of the record field? Handling this in pack/unpack seems clearer, and since we use an empty string as a marker in `pack()` it makes sense to me to do the same in `unpack()` for symmetry.

Reply:
This was necessarily changed as part of the type system refactoring. The `name` function returns a `std::string`, which may allocate, while `is_standalone_type` is guaranteed not to. 

```cpp
std::string qualified_record_field::name() const noexcept {
  if (layout_name_.empty())
    return std::string{field_.name};
  if (field_.name.empty())
    return std::string{layout_name_};
  return fmt::format("{}.{}", layout_name_, field_.name);
}

bool qualified_record_field::is_standalone_type() const noexcept {
  return field_.name.empty();
}
```

This is also visible in the documentation of the function:

```cpp
/// @note This function exists for backwards compatibility. While internally a
/// qualified record field is modeled as a record type with an index to a
/// known field, the partition synopsis used this data structure for both
/// field synopses as intended, and for type synopses which may be considered
/// a bad design decision. As a consequence, we need to be able to distinguish
/// whether a qualified record field is actually just a type as used for type
/// synopses.
[[nodiscard]] bool is_standalone_type() const noexcept;
```

Reply:
Do we have to consider that *all* types will soon have a non-empty `.name()`?

Reply:
> Do we have to consider that _all_ types will soon have a non-empty `.name()`?

I think we dropped the idea of doing that because of the many issues it caused, including breaking this.

For the next-gen partition FlatBuffers table I'd like to get rid of this hack entirely and store type and field synopses separately, instead of abusing a qualified record field without a name as essentially a type.